lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -116,7 +120,7 @@ def exit
         @stderr.close if !@stderr.closed?
 
         # wait up to 2 seconds for the watcher thread to fully exit
-        @ps_process.join(2)
+        @ps_process.join(2) if @ps_process.alive?
```

Comment:
I was going to say no, because:

`exit` is an instance method.  Each instance, upon creation, registers an `at_exit` handler to call `exit` for its instance.

The provider instances each ask for a `ps_manager` and then in `post_resource_eval` should clean it up... 

BUT

I just noticed that its `self.post_resource_eval`, so there should never be a `ps_manager` instance in scope.  In other words, `post_resource_eval` runs once for a type of provider, not per instance of provider.

So I may have to rethink this a bit... thanks!